src/ape/managers/project.py

```diff
@@ -86,30 +89,28 @@ def _extract_dependency_manifest(self, name: str, download_path: str) -> Package
                 manifest_dict = None
 
         if not manifest_dict:
-            manifest_dict = self._download_manifest(name, download_path, manifest_file_path)
+            manifest_dict = self._download_manifest(dependency_item, manifest_file_path)
 
         if "name" not in manifest_dict:
             manifest_dict["name"] = name.replace("_", "-")
 
         return PackageManifest(**manifest_dict)
 
-    def _download_manifest(self, name: str, download_path: str, manifest_target_path: Path) -> Dict:
+    def _download_manifest(
+        self, dependency_item: Union[IPFSDependency, GithubDependency], manifest_target_path: Path
+    ) -> Dict:
         manifest_dict = {}
+        name = dependency_item.name
 
-        if download_path.startswith("https://") or download_path.startswith("http://"):
-            response = requests.get(download_path)
+        if isinstance(dependency_item, IPFSDependency):
+            response = requests.get(dependency_item.ipfs)
             manifest_dict = response.json()
-        else:
-            # Github dependency (format: <org>/<repo>@<version>)
-            try:
-                path, version = download_path.split("@")
-            except ValueError:
-                raise ValueError("Invalid Github ID. Must be given as <org>/<repo>@<version>")
-
-            # Download manifest
+        elif isinstance(dependency_item, GithubDependency):
             with tempfile.TemporaryDirectory() as temp_dir:
                 temp_project_path = Path(temp_dir) / name
                 temp_project_path.mkdir(exist_ok=True, parents=True)
+                path = dependency_item.github
+                version = dependency_item.version
```

Comment:
Wonder if this could instead be some sort of `abstractmethod` in `DependencyConfig`, and then each dependency config type must implement the same method for fetching and extracting their content to a manfiest file

https://github.com/ApeWorX/ape/pull/431/files#diff-140ab16c725a530ed14f1900c24c98f2f3fc4b9ec4b96a0c439f0daeaf02de26R26